### DISCUSSION TITLE
Proper function of the CV Laser module (Ender 3 S1)

### DISCUSSION THREAD

**Discussion by PaWi20** [mriscoc/Ender3V2S1#1170] [Q&A]
Hey :) I have a few questions about the CV Laser module in combination with an Ender 3 S1 and the professional firmware by mriscoc (Ender3S1-F4-CVLM-20230904). 
First a few words for an introduction: I have decided to upgrade my Ender 3 S1 with the CV laser module. During installation, the motherboard of my printer broke. (I think I made the mistake of plugging the cable between the laser and the 24V connector while the printer was powered). I replaced the motherboard and did initial tests with the laser, but it didn't work as expected in the initial tests, but maybe it's just due to incorrect parameters for the process. I'm not sure if the laser module has a defect as well though. 

My questions relate more to the general procedure and the behavior of the laser hardware:
- Is it correct that the laser starts to light up as soon as "Toggle Laser" is activated? I assume that these are just laser beams with a very low power, right?
- Is this the usual procedure for laser processing: Place workpiece, set focus via Z-axis and T-bar spacer, select file, "Toggle Laser" (laser is switched on), Auto Home, (optional: Run Range), start engraving
- Can the Z-axis not be moved downwards via the laser menu, but only upwards?

I am very grateful for your feedback, especially on the first question.

Many thanks in advance,
Pascal

**Comment by TheWebMachine** [comment]
I have the CVLM and sponsor that version of the firmware. Happy to answer any questions I can about the laser module...it's an awesome piece of kit!

> Is it correct that the laser starts to light up as soon as "Toggle Laser" is activated? I assume that these are just laser beams with a very low power, right?

Yes, when you first enable `Toggle Laser`, a safe low-power dot will appear to aid in positioning and focus adjustments.

> Is this the usual procedure for laser processing: Place workpiece, set focus via Z-axis and T-bar spacer, select file, "Toggle Laser" (laser is switched on), Auto Home, (optional: Run Range), start engraving

I have found that the process works best as follows;
1) Place workpiece
2) `Toggle Laser` On
3) Use manual homing menu to set your desired origin relative to your workpiece
4) adjust Z focus with T-bar (this ensures focus relative to where you are engraving in case of slightly warped larger workpieces and yields better results in the finished piece
5) Load your gcode and use the `Run Range` option to confirm the area of work to be done
6) Engrave

>Can the Z-axis not be moved downwards via the laser menu, but only upwards?

That "bug" should have been addressed in the latest CVLM version (202030904). You should be able to take Z below 0 in the *laser* menu only.

I hope that helps!

**Reply by PaWi20** [reply]
Thank you so much for this quick response to my questions. That already helps a lot! 

> That "bug" should have been addressed in the latest CVLM version (202030904). You should be able to take Z below 0 in the laser menu only.

I'm pretty sure that I'm using this fw version. Currently I'm really busy for a few days, but I will check this as soon as I can.

So far it seems that everything should be fine with the laser module and that I didn't broke this too. But with my first tests last weekend I didn't manage to engrave anything. It seems that the process did stop after a few seconds. Although I could cut a line in thicker cardboard. 
I installed the origin laser firmware from Creality and received this message a few seconds after the laser process started: 
![CV Laser_Error](https://github.com/mriscoc/Ender3V2S1/assets/121219709/ad7ffa80-f72b-4ab1-bc0b-b9569b278d27)

Do you know what that could mean?

**Comment by TheWebMachine** [comment]
> I installed the origin laser firmware from Creality and received this message a few seconds after the laser process started:

That is due to the filament sensor, but I haven't used Creality's stock firmware in a long time now. Who knows what bugs it has. It was never very good firmware, hence the desire for truly professional firmware like this one.

**Reply by PaWi20** [reply]
Ok, fine. So another sign, that it's not my laser module. I already switched back to the professional firmware! :) As I said, I will do further tests by the end of the week.

Do you have some recommendations for the laser slicer settings (power, speed), let's say to engrave/cut paper and thin wood? I would like to use these settings to start off with.

**Reply by TheWebMachine** [reply]
Well, that is going to depend on a *lot* of factors. Even seemingly identical pieces of 3mm basswood are going to have slightly different densities between manufacturers, where a change of up to 20% power output could be needed.

What I did, for each new material I tried, was to leave slicer at default output and speed and did a simple small test image on the material to check how to adjust the power. Buy consistent brands of materials whenever possible to avoid frustration. Super cheap "craft" basswood is often super inconsistent, warps with the slightest humidity, and even warps from temp changes during the engraving process that can be enough to defocus and ruin the quality of the engraving (like looking at the image through rippling water).

Edit: Even be caution of more "exotic" cheap planks, as many are simply thin veneer of the expensive wood sandwiching cheap basswood in the middle, whereby engraving exposes the lower quality wood beneath that has a different burn rate. If it calls itself craft and is way cheaper than you'd expect, it's probable vaneer "plywood."

**Reply by PaWi20** [reply]
Many thanks again for this detailed information!

I have one last question for now about the topic exhaust air and emitted smoke/vapor during the process:
The laser module itself has an activated carbon filter. My printer itself is in a housing with a ventilated exhaust filter (activated carbon filter + HEPA (H13) filter). Its located in a not well ventilated room (no window, but with a room exhaust air, i.e. something like a bathroom fan). 
Do you think it's "safe" to proceed like this? I mean in theory the exhausted air out of the housing should be clean (2x carbon filter + HEPA).

Many thanks in advance for your time and advice!

**Reply by TheWebMachine** [reply]
No problem; glad to help.

The filter and fan on the laser itself is there mostly to draw vaporized particulate away from the piece to keep the laser focused - it only grabs about 50% of the air and only removes about 60% of the junk from that half of the air. That said, your enclosure ventilation will pick up the slack, but be prepared to change your enclosure filter often. That smoke is wood dust, wood oils, etc...a lot for the filter to collect up.

**Reply by PaWi20** [reply]
All right. So maybe I'll halve the time span of my filter change. It's hard to know when the filters are ready anyway as there is no "indicator". :D

I've also thought about placing the printer in front of our balcony door, leaving the housing door open and blowing the air out with a small fan when I'm doing "heavy" laser work.

**Comment by TheWebMachine** [comment]
> It's hard to know when the filters are ready anyway as there is no "indicator". :D

The smoke is so dark, I suspect it will make for its own indicator. haha  But the fan and door method will also work, weather permitting.

**Reply by PaWi20** [reply]
Hey @TheWebMachine :) 

I finally could test the laser module again and it works, also lowering the x-axis to adjust the focus.

I hope I can ask you some more questions that now came up:

Which subsurface do you use?
- I saw a video where someone put the stock print bed upside down, so that the aluminum surface is facing upwards. I'm really curious about this, as it will reflect the laser beam way more than the black "3d print" surface and can be a potential danger?
- I did some first tests with paper (engraving, cutting) and printed something like a honeycomb table out of TPU and put this on the print plate with the black side facing upwards. But of course I burned a little bit into TPU (and also the print bed).

Do you use the Creality Print software?
- What are the differences between the different "Processing Modes" for images?

I did some test engravings and the lines on the edges were way darker or even cut through the paper. I think the problem is at locations where the laser head is switching directions and therefore stays longer at a spot. How can I fix this?

Again many thanks in advance! :)

**Reply by TheWebMachine** [reply]
You'll have to excuse me as I haven't used my laser in a few months and I'm testing the latest FW version without CVLM support at the moment, so details are a little foggy for me...

> Which subsurface do you use?

I bought a laser engraving honeycomb to allow for properly clean cuts and airflow. You'll need to get one close to the size of your bed, but this is the one I bought, as it included pins to hold the work down. This one included an aluminum bottom plate, but you can use the printers plate flipped to the shiny side. The depth of the honeycomb defocuses the laser beneath the workarea, preventing reflections buring your workpiece or damaging the laser.

https://www.amazon.com/dp/B0BM443NS2

> Do you use the Creality Print software?

Yes, I do. This FW has been built and tested to accept all laser gcode as Creality generates it.

> What are the differences between the different "Processing Modes" for images?

This affects how the image is converted to an engraving. You can see how the different modes "dither" the image accordingly. Line Gray provides good results for actual images like photos. 

I've never really gotten into burning paper and felt, as those materials are quite fickle and not my project target, anyway. What I can tell you is that laser power and speed are companion parameters that will very much depend on material type and thickness. For each new type of material (or even different brands of the same material), I sacrifice a piece for testing. I print grayscale rectangles that go from white to black, black and white checkerboard squares, and plain square cut-outs with different power and speed settings until I get desired results. Then I can begin working with the material in earnest. There will be a lot of trial and error.

**Reply by PaWi20** [reply]
Thank you so much again for this quick, detailed information! :)

**Comment by TheWebMachine** [comment]
I just recalled that other software I use to generate laser gcode for vector objects (hard cuts, physical builds, etc) that doesn't require Creality's software. The best part is it is web-based, so it will run on any desktop OS (not mobile). It's great for creating your own cut outlines and such, which is something that Creality's software *can* do but not at all well...plus OpenBuilds CAM can do a few extra things like cut out letters in a variety of fonts and such.

https://cam.openbuilds.com/

When you first open it, it allows you to configure for your device. You will NOT use their CONTROL software and will ignore the first 3 steps completely. Then you select in Step 4 "Turn Laser On and Off: Constant Power (M3/M5)" as you will not likely be using this for variable engraving, like images and such (use Creality for that). Cutouts and basic vector drawings at a constant burn rate are where this software shines. Of course, if you'd like to give variable control a try and do more things like that, feel free to select Dynamic Power (M4/M5) instead.

Specify your printer's axis limits (300x,300y,300z in the case of my Plus), leave the Spindle command at S, set Power scale max to 1000, Rapid Mode G0, Linear Move G1, ***REMOVE*** G54 and G21 from Start G-Code. Here's a screenshot for example and these should be the only things you need to change on the Settings page:

![Screenshot from 2024-01-10 02-58-05](https://github.com/mriscoc/Ender3V2S1/assets/1045052/65bd9297-9eaa-4418-8f62-acd326dbf6fb)

Then you can Save and go play around! Once you have something laid out on the workarea, check one of the checkboxes for one of the vectors you want to generate (do one continuously cut object at a time for best results), then you click the green Add button that appears in Toolpaths to choose your parameters for that cut - select a Laser cut option depending on if you want the laser to cut inside, outside, or directly on the lines you've created (for accuracy for things like building objects, boxes, etc). The Power and Feedrate are all you need to adjust here for your workpiece based on the power and speed you need to cleanly cut or burn it. Once you have created the toolpaths for your objects, you can then click the Generate GCODE to actually create the GCODE file. Once G-code has been generated, the final bottom pane on the right will show a green Save GCODE button to export the gcode file so you can put it on SD and into the printer.

Sorry, that was not as brief a tutorial as I was hoping to write. The docs are pretty robust as this software is intended for everything from lasers to CNC, sprayers, and all sorts of CAM devices. It's just super handy for vector cuts. If you just want to cut a piece of wood into a circle real quick, this will get you from plotting to cutting in literally a couple of minutes. You can use it to pre- or post-cut something you engraved detail into with Creality's suite. It's a handy tool to have in your laser toolbox!
